tests/unit/test_credentials.py

```diff
@@ -720,6 +772,7 @@ def get_expected_creds_from_response(self, response):
             'secret_key': response['Credentials']['SecretAccessKey'],
             'token': response['Credentials']['SessionToken'],
             'expiry_time': expiration,
+            'account_id': None,
```

Comment:
Similar to the changed integration tests, this can be avoided if we only conditionally set `account_id` in the credential dictionary in the cached credential fetcher.  However, we haven't conditionally set values in the dict either even if the value is None so I elected not to.